### ISSUE TITLE
Secret store is unable to resolve an endpoint for oidc when using workload identity in azure: status code 404 with 0.10.6

### ISSUE THREAD

**Issue by papanito** [external-secrets/external-secrets#4139]
**Describe the bug**
We are using workload identity in azure for the secret store to authenticate against the keyvault. With the latest update, this does not work anymore and we get

```
unable to resolve an endpoint: http call(https://login.microsoftonline.com/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/oauth2/token/v2.0/.well-known/openid-configuration?)(GET) error: reply status code was 404
```

**To Reproduce**
1. provide all relevant manifests

  ```yaml
  apiVersion: v1
  kind: ServiceAccount
  metadata:
    annotations:
      azure.workload.identity/client-id: aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaa
    labels:
      azure.workload.identity/use: "true"
    name: sa-ess
    namespace: myns
  ---
  apiVersion: external-secrets.io/v1beta1
  kind: SecretStore
  metadata:
    name: secret-store-kv
    namespace:  myns
  spec:
    provider:
      azurekv:
        tenantId: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
        authType: WorkloadIdentity
        vaultUrl: "https://xxxx.vault.azure.net/"
        serviceAccountRef:
          name: sa-ess
  ```

2. provide the Kubernetes and ESO version
   
   Kubernetes 1.30.0

**Expected behavior**
No errors

**Screenshots**
N/A

**Additional context**

We upgraded from 0.10.5 to 0.10.6.
- helm chart 0.10.5 is working fine
- helm chart 0.10.6 breaks

**Comment by papanito** [comment]
Probably the same as https://github.com/external-secrets/external-secrets/issues/4135

**Comment by weisdd** [comment]
Looks like a duplicate of #4135, fixed by #4136. So, better to stick with 0.10.5 until a new version is released.

**Comment by Skarlso** [comment]
Please verify that the fix works, if yes, I'll do a patch release either tonight or tomorrow morning.

**Comment by weisdd** [comment]
@papanito I guess the easiest way for you would be to try `ghcr.io/external-secrets/external-secrets:main` (ofc, it's important to set `imagePullPolicy` to `Always` in case you've pulled an image with this tag before).

**Comment by Skarlso** [comment]
actually, each PR builds and publishes a version.
You can try this one: `ghcr.io/external-secrets/external-secrets:v0.10.6-4.ge7f74db3`

**Comment by Skarlso** [comment]
here are the images https://github.com/external-secrets/external-secrets/pkgs/container/external-secrets/versions?filters%5Bversion_type%5D=tagged&page=2

**Comment by andrew-liu1982** [comment]
facing the same 404 issue with Azure EntraID

**Comment by weisdd** [comment]
@andrew-liu1982 as it was mentioned above, you can try to run the image built off the main branch (`ghcr.io/external-secrets/external-secrets:main`) to see if it works for you.

**Comment by HujinoKun** [comment]
> ghcr.io/external-secrets/external-secrets:main

Hello,

It is better with this image, thx

**Comment by Skarlso** [comment]
I'll do a release later on.

**Comment by gusfcarvalho** [comment]
closed as duplicate of #4135
